src/prelude/internals.mo

```diff
@@ -612,9 +612,22 @@ func @timer_helper() : async () {
   ignore (prim "global_timer_set" : Nat64 -> Nat64) exp;
   if (exp == 0) @timers := null;
 
+  var failed : Nat64 = 0;
+  func reinsert(job : () -> async ()) {
+    if (failed == 0) @timers := @prune @timers;
+    failed += 1;
+    @timers := ?(switch @timers {
+      case (?{ id = 0; pre; post; job = j; expire; delay })
```

Comment:
why does this match on id =  0?

Reply:
Only already `reinsert`-ed nodes (special `id` = 0) should be delegated into the `pre` part of the PQ.

Reply:
This tree rotation maintains the original expiration order.